modules/front-end/src/app/core/components/find-rule/find-rule.component.html

```diff
@@ -27,7 +27,7 @@
 
   <ng-template #titleTemplate>
     <div style="display:flex;justify-content: flex-start;">
-      <input type="text" class="standard-bordered-input" nz-input style="width: 500px;height: 36px;" [(ngModel)]="name" (ngModelChange)="onRuleNameChange()" />
+      <input type="text" class="standard-bordered-input" nz-input style="width: 500px;height: 36px;" [formControl]="nameControl" />
```

Comment:
This change is unnecessary since the input is not under a form element.

Reply:
You may have some misunderstanding here, we can use `ngModelChange` without a formControl. The orginal implementation is fine.

Good articles to read about

- https://www.angularjswiki.com/angular/ngmodelchange-change-angular: Understand `ngModelChange` section
- https://ultimatecourses.com/blog/difference-change-ngmodelchange-angular